File: practice-3/rust-excerices/src/check_balance.rs
use solana_client::rpc_client::RpcClient;
use solana_sdk::pubkey::Pubkey;
use std::str::FromStr;

const LAMPORTS_PER_SOL: f64 = 1_000_000_000.0;

pub fn get_connection() -> RpcClient {
    let cluster = "https://api.devnet.solana.com";
    RpcClient::new(cluster.to_string())
}

pub fn check_balance(pubkey: &str) {
    let connection = get_connection();
    let public_key = Pubkey::from_str(pubkey).expect("Invalid PUB_KEY");
    let balance_in_lamports = connection
        .get_balance(&public_key)
        .expect("Failed to get balance");
    let balance_in_sol = balance_in_lamports as f64 / LAMPORTS_PER_SOL;
    println!("Public Key: {pubkey}: Balance: {balance_in_sol} SOL");
}

File: practice-2/type_script/src/utils/check-balance.ts
import { Connection, LAMPORTS_PER_SOL } from "@solana/web3.js";
import { getPublicKey } from "../utils";

const publicKey = getPublicKey();

export const checkBalance = async (connection: Connection) => {
  const balanceInLamports = await connection.getBalance(publicKey);
  const balanceInSOL = balanceInLamports / LAMPORTS_PER_SOL;
  console.log(`Key: ${publicKey}; Balance: ${balanceInSOL}SOL`);
};
